crates/holochain_state/src/source_chain.rs

```diff
@@ -381,6 +410,46 @@ impl SourceChain {
         Ok(elements)
     }
 
+    pub async fn unlock_chain(&self) -> SourceChainResult<()> {
+        self.vault
+            .async_commit(move |txn| unlock_chain(txn))
+            .await?;
+        Ok(())
+    }
+
+    pub async fn accept_countersigning_preflight_request(
+        &self,
+        preflight_request: PreflightRequest,
+        agent_index: u8,
+    ) -> SourceChainResult<CounterSigningAgentState> {
+        let hashed_preflight_request = holo_hash::encode::blake2b_256(
+            &holochain_serialized_bytes::encode(&preflight_request)?,
+        );
+
+        // This all needs to be ensured in a non-panicky way BEFORE calling into the source chain here.
+        let author = self.author.clone();
+        assert!(*author == preflight_request.signing_agents()[agent_index as usize].0);
```

Comment:
Is there any reason to use an assert vs an error here?

Reply:
@freesig because it's a compile time error not a runtime issue, this check is duplicated in the calling function

Reply:
and by compile time error i just mean a logic bug, there's no case where this can work

Reply:
i can do `assert_eq`